python/pysrc/kaskada/_timestream.py

```diff
@@ -880,6 +906,71 @@ def record(self, fields: Callable[[Timestream], Mapping[str, Arg]]) -> Timestrea
         """
         return record(fields(self))
 
+    def round(self) -> Timestream:
+        """Return a Timestream with all values rounded to the nearest integer.
+
+        Returns:
+            A Timestream of the same type as `self`. The result contains `null`
+            if the value was `null` at that point. Otherwise, it contains
+            the result of rounding the value to the nearest integer.
+
+        Notes:
+            This method may be applied to any numeric type. For anything other
+            than `float32` and `float64` it has no affect since the values
+            are already integers.
+
+        See Also:
+            :func:`ceil`
+            :func:`floor`
+        """
+        return Timestream._call("round", self)
+
+    def sqrt(self) -> Timestream:
+        """Return a Timestream with the square root of all values."""
+        return Timestream._call("sqrt", self)
+
+    def upper(self) -> Timestream:
+        """Return a Timestream with all values converted to upper case."""
+        return Timestream._call("upper", self)
+
+    def greatest(self, rhs: Union[Timestream, Literal]) -> Timestream:
+        """Return a Timestream with the greatest value of `self` and `rhs` at each point.
+
+        Args:
+            rhs: The Timestream or literal value to compare to this.
+
+        Returns:
+            Each point contains the value from `self` if `self`
+            is greater than `rhs`, otherwise it contains `rhs`.
+            If any input is `null` or `NaN`, then that will be
+            the result.
+
+        See Also:
+            This returns the greatest of two values. See the
+            aggregation :func:`max` for the maximum of values in
+            a column up to and including the current row.
```

Comment:
Suggest dropping the "up to and including the current row" since that will depend on the windowing, etc.